### PR TITLE
Add `pre-commit` hooks

### PR DESCRIPTION
**Pull request by ndmlny-qs**
This commit adds the ability to run `pre-commit` on code committed to the repo. I have run it on the repo, which has required the following files to be updated.

- CONTRIBUTING.md
- README.md
- arviz_dashboard/__init__.py
- arviz_dashboard/elpd.py
- examples/elpd_dashboard.ipynb
- examples/posterior-marginal1d-examples.ipynb

There are several configurations we can use for `pre-commit` that can be stored in a `pyproject.toml` file. This commit also updates the `setup.py` file so we can install locally using the `pyproject.toml` file instead of the legacy `setup.py` file.

### CODE REVIEW

**Comment by ndmlny-qs** [comment]
@aloctavodia this is a large change, but will help keep code hygiene for future contributors. I will need to add documentation on how to use `pre-commit`, but once you understand what it is doing you will (hopefully) like it.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    hooks:
+      - id: dead
+        name: remove commented out code
+        description: python dead code detection
+        entry: dead
+        pass_filenames: false
+        language: python
+        language_version: python3
+        types:
+          - python
```
Comment (aloctavodia) [inline .pre-commit-config.yaml:116]: this seems to be duplicated

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+[project.optional-dependencies]
+dev = ["black", "flake8", "isort", "pre-commit", "pyre-check", "ruff"]
+
+[project.urls]
+"Bug Tracker" = "https://github.com/arviz-devs/arviz_dashboard/issues"
+Homepage = "https://www.arviz.org"
+"Source Code" = "https://github.com/arviz-devs/arviz_dashboard"
+
+[tool.flake8]
+max-line-length = 88
```
Comment (aloctavodia) [inline pyproject.toml:38]: Nitpick, for other projects we use 100

**Comment by ndmlny-qs** [comment]
closing as a newer branch has been worked on for the issue

**Pull request closed without merging**